tendrl/ceph_integration/manager/crud.py

```diff
@@ -119,3 +121,14 @@ def delete(self, object_type, object_id):
             )
         else:
             raise NotImplementedError(object_type)
+
+    def sync_request_status(self, request):
+        count = 0
+        while count < 30:
+            gevent.sleep(10)
+            if request.state == "complete" and not request.error:
+                return 0, request.status
+            if request.error:
+                return -1, request.error_message
+            count += 1
+        return -1, "Request timed out"
```

Comment:
please stop using these exit code, status style of programming.  It is always better to raise custom exceptions on failures and use try except to call sync_request_status 

More reading: https://jeffknupp.com/blog/2013/02/06/write-cleaner-python-use-exceptions/